src/Controller/Component/PaginatorComponent.php

```diff
@@ -187,9 +187,10 @@ public function paginate($object, array $settings = [])
             $query->applyOptions($options);
         }
 
+        $cleanQuery = clone $query;
         $results = $query->all();
         $numResults = count($results);
-        $count = $numResults ? $query->count() : 0;
+        $count = $numResults ? $cleanQuery->count() : 0;
```

Comment:
count() is supposed to do the cloning. Should we fix it there? Or maybe remove it?